### ISSUE TITLE
regression: nocarp no longer working since 2.0.1

### ISSUE THREAD

**Issue by boudekerk** [gnustavo/Git-Hooks#39]
After upgrading to 2.x, setting nocarp to 1 has no effect anymore:
```
remote: 
remote: [Git::Hooks::CheckJira] the JIRA issue(s) (TEST-152) do(es) not match the expression 'key IN (TEST-152) AND project IN ('HELPDESK')'
remote:  at /data/git/repositories/sandbox/bartosz.git/custom_hooks/pre-receive line 4
```
This worked with the 1.16.0 and doesn't for any version since 2.0.1, including the latest 2.1.3

**Comment by boudekerk** [comment]
Just to add, it is enabled via the githooks.nocarp option as indicated by the Changelog/documentation.

**Comment by boudekerk** [comment]
Updated my comment and the title, as I haven't actually tested with 2.0.0

**Comment by gnustavo** [comment]
Hi. I think I fixed it in v2.1.6 which I just released. Please try it out and let me know if it works for you.

Thank you for the heads up!

**Comment by boudekerk** [comment]
I think the behaviour is reversed now. We now never get the suffix, regardless of the nocarp setting.
Which 'works for me', as I can now deploy it to production, but I guess it's still not the correct behaviour.

**Comment by gnustavo** [comment]
Hi. It took me a while to be able to get back to this.

As I payed more attention to the issue I realized that the nocarp directive doesn't make sense in GH version 2 because the hooks now don't signal errors by dying but by registering error messages via the error method and returning 0. So, their messages shouldn't have line numbers inserted by Perl in any case.

So, I removed the directive from the documentation and from the code altogether in c6e08f1.

OK?
